Developer Guide (#1817)
arviz-devs/arviz: doc/source/contributing/developer_guide.md

# Developer Guide
## Library architecture
ArviZ is organized in modules (the folders in [arviz directory](https://github.com/arviz-devs/arviz/tree/main/arviz)).
The main 3 modules are `data`, `plots` and `stats`.

> The [tests](https://github.com/arviz-devs/arviz/tree/main/arviz/tests) folder contains tests for all these 3 modules.

Reviewer: ```suggestion
Then we have 3 more folders. The [tests](https://github.com/arviz-devs/arviz/tree/main/arviz/tests) folder contains tests for all these 3 modules.
```

> [static](https://github.com/arviz-devs/arviz/tree/main/arviz/static) is only used to store style and CSS files to get HTML output for `InferenceData`.
> [wrappers](https://github.com/arviz-devs/arviz/tree/main/arviz/wrappers) is still experimental (not tested yet either) and interacts closely with both [data](https://github.com/arviz-devs/arviz/tree/main/arviz/data) and [stats](https://github.com/arviz-devs/arviz/tree/main/arviz/stats) modules.

Reviewer: ```suggestion
The [static](https://github.com/arviz-devs/arviz/tree/main/arviz/static) folder is only used to store style and CSS files to get HTML output for `InferenceData`. Finally we have the [wrappers](https://github.com/arviz-devs/arviz/tree/main/arviz/wrappers) folder. Which contains experimental (not tested yet either) features and interacts closely with both [data](https://github.com/arviz-devs/arviz/tree/main/arviz/data) and [stats](https://github.com/arviz-devs/arviz/tree/main/arviz/stats) modules.
```

Reviewer: This and the previous suggestion should be part of the same paragraph. (github did not allowed me to create a single suggestion)

In addition, there are some files on the higher level directory: `utils.py`, `sel_utils.py`,
`rcparams.py` and `labels.py`.

> ## plots

Reviewer: ```suggestion
## Plots
```

> ArviZ now supports multiple backends. While adding another backend, please ensure you meet the

Reviewer: ```suggestion
ArviZ supports multiple backends. While adding another backend, please ensure you meet the
```

Sphinx is configured to ease referencing libraries ArviZ relies heavily on by
using [intersphinx](https://docs.readthedocs.io/en/stable/guides/intersphinx.html).
See guidance on the reference about how to link to objects from external

> libraries and the value of [intersphinx_mapping](https://github.com/arviz-devs/arviz/blob/main/doc/source/conf.py#L270) in [conf.py](https://github.com/arviz-devs/arviz/blob/main/doc/source/conf.py#L270) for the complete and up to

Reviewer 2: Having links to the files on github is ok (I don't think it's necessary unless there could be some confusion on the file otherwise but it clearly doesn't hurt) however, I think we should avoid links with line numbers as they quickly stop pointing at the right place and end up being more confusing than having readers search for a variable or piece of code manually
